app.rb

```diff
@@ -1,8 +1,94 @@
+# in app.rb
+
 require 'sinatra/base'
-class RockPaperScissors < Sinatra::Base
-  get '/test' do
-    'test page'
+require_relative './lib/game'
+require_relative './lib/player'
+
+class RPSLS < Sinatra::Base
+
+  # Allows us to save between requests
+  enable :sessions
+
+
+  get '/' do
+    if (defined?(@@game)).nil? then @@game = Game.new end
+    erb :index
+  end
+
+  post '/reset' do
+    @@game = Game.new
+    redirect '/'
+  end
+
+  post '/action' do
+    if params[:player] == "p1" then @@game.player_1.selection = params[:action] end
+    if params[:player] == "p2" then @@game.player_2.selection = params[:action] end
+    redirect '/waiting-room-game'
+  end
+
+  post '/waiting-room-1' do
+    @@game.players[0] = Player.new(params[:player_1_name])
+    redirect '/waiting-room-names'
+  end
+
+  post '/waiting-room-2' do
+    @@game.players[1] = Player.new(params[:player_2_name])
+    redirect '/waiting-room-names'
+  end
+
+
+  get '/waiting-room-names' do
+    print @@game.players
+    erb :waiting_room_names
   end
 
+  get '/waiting-room-game' do
+    print "P1 Action = #{@@game.player_1.selection}"
+    print "P2 Action = #{@@game.player_2.selection}"
+    erb :waiting_room_game
+  end
+
+  # in app.rb
+  get '/play' do
+    @game = @@game
+    erb :play
+  end
+
+  # in app.rb
+  get '/results' do
```

Comment:
Great use of Single Responsibility Principle throughout, all methods are very concise and easy to read!